packages/new-polymath-sdk/src/Polymath.ts

```diff
@@ -1,53 +1,71 @@
 import { HttpProvider, WebsocketProvider } from 'web3/providers';
-import { types, constants } from '@polymathnetwork/new-shared';
+import { BigNumber } from 'bignumber.js';
 import { PolyToken } from '~/LowLevel/PolyToken';
-import { Erc20 } from '~/LowLevel/Erc20';
 import { LowLevel } from '~/LowLevel';
 import { PolymathRegistry } from '~/LowLevel/PolymathRegistry';
 import { SecurityTokenRegistry } from '~/LowLevel/SecurityTokenRegistry';
-import { Wallet } from './Wallet';
+import { Context } from '~/Context';
+import { ModuleRegistry } from '~/LowLevel/ModuleRegistry';
+import { TaxWithholding } from '~/types';
+import { Dividend } from '~/LowLevel/types';
+import { SecurityToken } from '~/entities';
 import {
   ReserveSecurityToken,
   EnableDividendModules,
   CreateCheckpoint,
   CreateErc20DividendDistribution,
   CreateEtherDividendDistribution,
-} from './transactions';
-import { Context } from './Context';
-import BigNumber from 'bignumber.js';
-import { ModuleRegistry } from '~/LowLevel/ModuleRegistry';
-import { TaxWithholding } from '~/types';
-import { Dividend } from '~/LowLevel/types';
+} from './procedures';
+import { CreateSecurityToken } from '~/procedures/CreateSecurityToken';
+import * as entities from '~/entities';
 
-interface Params {
+export type EntityClasses = typeof entities;
+
+export interface PolymathNetworkParams {
   httpProvider?: HttpProvider;
   httpProviderUrl?: string;
   wsProvider?: WebsocketProvider;
   wsProviderUrl?: string;
   polymathRegistryAddress: string;
 }
 
-export class PolymathClient {
+const createContextualizedEntity = (
+  ClassToContextualize: EntityClasses,
+  polyClient: Polymath
+) => {
+  class ContextualizedEntity extends ClassToContextualize {
+    constructor(params: any) {
+      super(params, polyClient);
+    }
+  }
+  return ContextualizedEntity;
+};
+
+// TODO @RafaelVidaurre: Fix typing here
+interface ContextualizedEntityClasses {
+  SecurityToken: any;
```

Comment:
Not relevant yet, will add these as they are exported to the public API. Halting the progress on entities for now though. Commented out the code because of typing issues